### PR TITLE
chore: logging 2 – replace print in analysis_db files

### PR DESCRIPTION
**Pull request by hollandjg**
builds on
- #175

### CODE REVIEW

File: src/icesat2_tracks/analysis_db/B04_angle.py [hunk 1]
```diff
@@ -280,7 +279,7 @@ def run_B04_angle(
             ),
         }
         MT.json_save("B04_fail", plot_path, dd_save)
-        print("exit()")
+        _logger.critical("exit()")
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B04_angle.py:282]:
```suggestion
        _logger.critical("terminating..)")
```

File: src/icesat2_tracks/analysis_db/B04_angle.py [hunk 2]
```diff
@@ -556,12 +555,12 @@ def plot_instance(
     ggg, xxx = np.meshgrid(group_number, x_list.data)
 
     for gi in zip(ggg.flatten(), xxx.flatten()):
-        print(gi)
+        _logger.debug(gi)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B04_angle.py:558]:
```suggestion
        _logger.debug("gi = %s", gi)
```

File: src/icesat2_tracks/analysis_db/B04_angle.py [hunk 3]
```diff
@@ -842,7 +841,7 @@ def get_instance(k_pair):
         L_optimize = L_optimize.T.sort_values("K_prime")
         L_brute = L_brute.T.sort_values("K_prime")
 
-        print("done with ", group, xi / 1e3)
+        _logger.info("done with %s %s", group, xi / 1e3)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B04_angle.py:844]:
```suggestion
        _logger.info("done with group %s,  xi = %s", group, xi / 1e3)
```

File: src/icesat2_tracks/analysis_db/B04_angle.py [hunk 4]
```diff
@@ -880,7 +879,7 @@ def get_instance(k_pair):
             {"L_sample": LL}, save_name + "_res_table", str(save_path)
         )  # TODO: clean up save_pandas_table to use pathlib
     except Exception as e:
-        print(f"This is a warning: {e}")
+        _logger.warning(f"This is a warning: %s", e)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B04_angle.py:882]:
```suggestion
        _logger.warning("This is a warning: %s", e)
```

File: src/icesat2_tracks/analysis_db/B05_define_angle.py [hunk 5]
```diff
@@ -119,8 +120,8 @@ def __init__(
 
         self.min = np.round(np.nanmin(data[freq_sel_bool, :]), 2)
         self.max = np.round(np.nanmax(data[freq_sel_bool, :]), 2)
-        if verbose:
-            print(str(self.min), str(self.max))
+
+        _logger.info("%s %s", str(self.min), str(self.max))
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B05_define_angle.py:124]:
```suggestion
        _logger.info("min %s, max %s", self.min, self.max)
```

File: src/icesat2_tracks/analysis_db/B05_define_angle.py [hunk 6]
```diff
@@ -427,7 +428,7 @@ def define_angle(
                 "reason": "not enough x segments",
             },
         )
-        print("exit()")
+        _logger.critical("exit()")
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B05_define_angle.py:431]:
```suggestion
        _logger.critical("exiting")
```

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 7]
```diff
@@ -77,18 +77,18 @@ def get_correct_breakpoint(pw_results):
...
-        echo("no negative slope found, set to lowest")
+        _logger.debug("no negative slope found, set to lowest")
         breakpoint = "start"
     else:
         # take steepest slope
         alpah_v2_sub = alphas_v2_sorted[slope_mask]
-        echo(alpah_v2_sub)
-        echo(alpah_v2_sub.argmin())
+        _logger.debug("%s", alpah_v2_sub)
+        _logger.debug("%s", alpah_v2_sub.argmin())
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:91]:
```suggestion
        _logger.debug("alpah_v2_sub: %s", alpah_v2_sub)
        _logger.debug("alpah_v2_sub.argmin: %s", alpah_v2_sub.argmin())
```

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 8]
```diff
@@ -105,7 +105,9 @@ def get_breakingpoints(xx, dd):
     n_breakpoints = 3
     while convergence_flag:
         pw_fit = piecewise_regression.Fit(xx, dd, n_breakpoints=n_breakpoints)
-        print("n_breakpoints", n_breakpoints, pw_fit.get_results()["converged"])
+        _logger.debug(
+            "n_breakpoints %s %s", n_breakpoints, pw_fit.get_results()["converged"]
+        )
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:110]:
```suggestion
        _logger.debug(
            "n_breakpoints %s\n%s", n_breakpoints, pw_fit.get_results()["converged"]
        )
```

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 9]
```diff
@@ -165,12 +167,12 @@ def weighted_mean(data, weights, additional_data=None):
 
 
 def calculate_k_end(x, k, k_end_previous, G_gFT_smth):
-    echo(x)
+    _logger.debug("%s", x)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:170]:
```suggestion
    _logger.debug("From calculate_k_end -- x: %s", x)
```

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 10]
```diff
@@ -657,7 +663,7 @@ def reconstruct_displacement(Gx_1, Gk_1, T3, k_thresh):
     Gx_v2, B2_v2, B3_v2 = dict(), dict(), dict()
 
     for bb in Gx.beam.data:
-        echo(bb)
+        _logger.debug("%s", bb)
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:666]:
```suggestion
        _logger.debug("bb: %s", bb)
```

File: src/icesat2_tracks/analysis_db/B06_correct_separate_var.py [hunk 11]
```diff
@@ -773,11 +779,12 @@ def reconstruct_displacement(Gx_1, Gk_1, T3, k_thresh):
         (plot_path / "../"),
         {"time": time.asctime(time.localtime(time.time()))},
     )
-    echo("done. saved target at " + str(plot_path) + "../B06_success")
-    echo("Done B06_correct_separate_var")
+    _logger.info("done. saved target at %s../B06_success", str(plot_path))
```
Comment (cpaniaguam) [inline src/icesat2_tracks/analysis_db/B06_correct_separate_var.py:782]:
```suggestion
    _logger.info("done. saved target at %s../B06_success", plot_path)
```

**Review by cpaniaguam — CHANGES_REQUESTED** [review]
A few suggestions here.

**Review by cpaniaguam — APPROVED** [review]
👍

File: src/icesat2waves/analysis_db/B06_correct_separate_var.py [hunk 12]
```diff
@@ -165,12 +167,12 @@ def weighted_mean(data, weights, additional_data=None):
...
 
 def calculate_k_end(x, k, k_end_previous, G_gFT_smth):
-    echo(x)
+    _logger.debug("From calculate_k_end -- x: %s", x)
     k_end, _ = define_noise_wavenumber_piecewise(
         G_gFT_smth.sel(x=x) / k, plot_flag=False
     )
     k_save = k_end_previous if k_end == k[0] else k_end
-    echo("--------------------------")
+    _logger.debug("--------------------------")
```
Comment (kmilo9999) [inline src/icesat2waves/analysis_db/B06_correct_separate_var.py:175]: It's this print important ?

**Review by kmilo9999 — DISMISSED** [review]
Looks good

**Review by cpaniaguam — APPROVED** [review]

**Pull request merged**